### ISSUE TITLE
Added Counting Sort.cpp

### ISSUE THREAD

**Issue by samyak06** [hacktoberfest2k20/DataStructures-and-Algorithms#46]
Please assign this to me.
I am new to open source contributions.
I have opened a PR for this #45 
Thank You.

**Comment by s-ayush2903** [comment]
Yeah, I see, can you please link this issue with the PR you opened

**Comment by samyak06** [comment]
I have linked the issue with the PR.
